numpy/_core/strings.py

```diff
@@ -137,17 +137,35 @@ def multiply(a, i):
     >>> np.strings.multiply(a, i)
     array([['a', 'bb', 'ccc'],
            ['d', 'ee', 'fff']], dtype='<U3')
-           
+
     """
-    a_arr = np.asarray(a)
-    i_arr = np.asarray(i)
-    if not issubclass(i_arr.dtype.type, np.integer):
-        raise ValueError("Can only multiply by integers")
-    if a_arr.dtype.kind == "T":
-        return a_arr * i_arr
-    out_size = _get_num_chars(a_arr) * max(int(i_arr.max()), 0)
-    return _vec_string(
-        a_arr, type(a_arr.dtype)(out_size), '__mul__', (i_arr,))
+    a = np.asanyarray(a)
+    i = np.asanyarray(i)
+    a, i = np.broadcast_arrays(a, i)
+    if not np.issubdtype(i.dtype, np.integer):
+        raise TypeError(f"unsupported type {i.dtype} for operand 'i'")
+
+    # (negative number) * (any string dtype) should return an empty string
+    i = np.where(i < 0, 0, i)
+
+    # delegate to stringdtype loops that also do overflow checking
+    if a.dtype.char == "T":
+        return a * i
+
+    # Check whether sizeof(buffer) > sys.maxsize avoiding overflow
+    # and division by zero
+    a_len = str_len(a)
```

Comment:
I think this can be simplified to,
```
a_len = str_len(a)
if np.any(a_len > sys.maxsize / np.maximum(i, 1)):
```
The advantage of not broadcasting before is that for the typical case of `i` being a fixed number, this is much faster.